tests/integrations/test_postcode_lookup_helper.py

```diff
@@ -0,0 +1,100 @@
+import requests
+import json
+from unittest import mock
+from flask import Flask
+
+from vulnerable_people_form.integrations.postcode_lookup_helper import \
+        entry_is_a_postal_address, \
+        get_addresses_from_postcode, \
+        PostcodeNotFound, ErrorFindingAddress
+
+_current_app = Flask(__name__)
+
+
+def test_entry_is_a_postal_address_function(fake_os_places_api_entry):
+    postable_entry = fake_os_places_api_entry(postal_address_code='D')
+    non_postable_entry = fake_os_places_api_entry(postal_address_code='N')
+
+    assert entry_is_a_postal_address(postable_entry.to_json()) is True
+    assert entry_is_a_postal_address(non_postable_entry.to_json()) is False
+
+
+@mock.patch('requests.get')
+def test_get_addresses_from_postcode(mock_get, faker, fake_os_places_api_entry, fake_os_places_api_response):
+    postcode = faker.postcode()
+
+    entries = []
+    # GIVEN 5 postable address entries for a postcode
+    for _ in range(0, 5):
+        entries.append(fake_os_places_api_entry(postcode=postcode, postal_address_code='D'))
+    # and one non-postable entry.
+    entries.append(fake_os_places_api_entry(postcode=postcode, postal_address_code='N'))
+
+    fake_api_response = fake_os_places_api_response(
+        postcode=postcode,
+        entries=entries,
+    )
+    mock_get.return_value = create_response_object(200, fake_api_response.to_json())
+
+    # WHEN we call the get_addresses_from_postcode function
+    with _current_app.test_request_context():
+        addresses = get_addresses_from_postcode(fake_api_response.postcode)
+
+        # THEN we get 5 postable address entries back with the details we expect.
+        assert len(addresses) == 5
+        api_response_and_function_result_should_match(fake_api_response, addresses)
+
+
+@mock.patch('requests.get')
+def test_get_addresses_from_postcode_handles_400_status_code(mock_get, faker):
+    postcode = faker.postcode()
+    mock_get.return_value = create_response_object(400, '')
+    with pytest.raises(PostcodeNotFound):
+        with _current_app.test_request_context():
+            get_addresses_from_postcode(postcode)
+
+
+@mock.patch('requests.get')
+def test_get_addresses_from_postcode_handles_401_status_code(mock_get, faker):
+    postcode = faker.postcode()
+    mock_get.return_value = create_response_object(401, '')
+    with pytest.raises(ErrorFindingAddress):
+        with _current_app.test_request_context():
+            get_addresses_from_postcode(postcode)
+
+
+@mock.patch('requests.get')
+def test_get_addresses_from_postcode_handles_500_status_code(mock_get, faker):
+    postcode = faker.postcode()
+    mock_get.return_value = create_response_object(500, '')
+    with pytest.raises(ErrorFindingAddress):
+        with _current_app.test_request_context():
+            get_addresses_from_postcode(postcode)
+
+
+def api_response_and_function_result_should_match(api_response, addresses):
+    for i in range(0, len(api_response.address_entries)):
```

Comment:
I think it might be better to filter out the unpostable addresses before calling this function, rather than having a switch that ignores them? Maybe it doesn't make much difference, but it feels like you're relying on the indexes still matching after you skip over 'N' entries and other things, seems like it could be brittle if the tests had to be extended much.

Reply:
I agree, i think it can be merged without this but would be something to look into if there is time

Reply:
I've changed the test to exclude the non-postable entry from the list of expected entries passed to the compare function. It also now asserts the number of addresses returned is equal to the number of expected entries. Is this better?

Reply:
works for me! 👍